api/server.py

```diff
@@ -104,7 +142,7 @@ def add_custom_headers(response):
 
     # Access-Control-Allow-Origin for CORS
     response.headers["Access-Control-Allow-Origin"] = "*"
-    response.headers["Access-Control-Allow-Methods"] = "GET, POST, OPTIONS"
+    response.headers["Access-Control-Allow-Methods"] = "GET, POST, PUT, OPTIONS"
```

Comment:
i spent a long time debugging this, and i could not figure out why the `update_user` API is not running into a CORS issue in prod. if you want to go down the same rabbit hole I did, undo this change and compare the CORS error you get locally vs prod 😵‍💫

Reply:
Yech, this is on me. 

I touched on it in the API gateway skill share but only for a few seconds. The API gateway is rewriting the CORS headers, and I have it set to wide open. 

Like I said in a comment above (or below, you never know with GH) -- I default to the wildcard, but I am pretty sure it should be the list of URL hostnames who are allowed to access the resource. I never quite get it, because I figure you can defeat it by changing how your browser works -- I don't understand CORS, beyond the most basic thing, if that.

Reply:
Ah, of course! I completely forgot about the gateway. I'm so glad this is making sense now. 

Hmm, yes, I'm with you—I think wildcard is definitely fine for verbs. Maybe we should go ahead and whitelist our referrers?

In terms of the headers that this API is sending, I guess we should try to make these match the gateway exactly, and I'll add a code comment that mentions that these may be overwritten in the production deployment?

Reply:
That's 100% cool, and for the CORS headers -- you also can just flat ignore them. Maybe leave a comment that there is an API gateway in use and it handles it.

I'm fairlyh strongly confident that nothing the app does will change what CORS behavior we exhibit on the other side of the gateway.

Reply:
> I'm fairly strongly confident that nothing the app does will change what CORS behavior we exhibit on the other side of the gateway.

Well, we're stuck in this awkward spot where we need to local API to work without the gateway, so those allowed methods definitely need to be right 🙃 

I dunno—maybe it would be better to dedupe the header setting and put them all in our flask app? i don't think it's a big deal to manage them in two places, so long as we add comments here as a warning...

Reply:
Oh! Of course - I fully hadn't considered that it's got to run locally too. 🙃 

Yes - everything you're saying makes perfect sense now and I am entirely cool with however you feel best to go forward -- 100%.

Reply:
Cool—I added code comments to remind us that the headers may be overwritten by our gateway 🙏